### PR TITLE
[et] Better RBE defaults

### PR DESCRIPTION
**Pull request by zanderso**
This PR adopts some RBE configuration from the way that chromium uses RBE

https://chromium.googlesource.com/chromium/tools/depot_tools.git/+/refs/heads/main/reclient_helper.py

These changes should bias both local and CI builds more towards using the worker pool, which we recently expanded, and should help limit the bandwidth used, which is a bottleneck for build times on a slow connection.

### CODE REVIEW

File: tools/gn [hunk 1]
```diff
@@ -226,6 +226,10 @@ def setup_rbe(args):
   # care of starting and stopping the compiler proxy.
   running_on_luci = os.environ.get('LUCI_CONTEXT') is not None
 
+  # Don't use racing on CI.
```
Comment (cbracken) [inline tools/gn:229]: Are there docs that explain what racing means in the context of RBE? Would be handy to link them here maybe.

Comment (cbracken) [reply tools/gn:229]: Or add a See: RbeConfig enum maybe?

File: tools/pkg/engine_build_configs/test/build_config_runner_test.dart [hunk 2]
```diff
@@ -283,6 +283,135 @@ void main() {
...
+      events[4].environment!.containsKey('RBE_local_resource_fraction'),
+      isTrue,
+    );
+    expect(
+      events[4].environment!['RBE_local_resource_fraction'],
+      equals('0.2'),
+    );
+  });
+
+  test('GlobalBuildRunner sets RBE_disable_remote when appropriate', () async {
```
Comment (cbracken) [inline tools/pkg/engine_build_configs/test/build_config_runner_test.dart:332]: `s/when appropriate/when $CONDITION/`

File: tools/pkg/engine_build_configs/test/build_config_runner_test.dart [hunk 3]
```diff
@@ -283,6 +283,135 @@ void main() {
...
+    expect(events[4].environment!['RBE_remote_disabled'], equals('1'));
+    expect(events[4].environment!.containsKey('RBE_exec_strategy'), isFalse);
+    expect(events[4].environment!.containsKey('RBE_racing_bias'), isFalse);
+    expect(
+      events[4].environment!.containsKey('RBE_local_resource_fraction'),
+      isFalse,
+    );
+  });
+
+  test('GlobalBuildRunner sets RBE_exec_strategy when appropriate', () async {
```
Comment (cbracken) [inline tools/pkg/engine_build_configs/test/build_config_runner_test.dart:372]: Ditto here.

**Review by cbracken — APPROVED** [review]
lgtm modulo tiny nits!

LGTM stamp from a Japanese personal seal

File: tools/gn (hunk 1 above)
Comment (zanderso) [reply tools/gn]: Done.

File: tools/pkg/engine_build_configs/test/build_config_runner_test.dart (hunk 2 above)
Comment (zanderso) [reply tools/pkg/engine_build_configs/test/build_config_runner_test.dart]: Done.

File: tools/pkg/engine_build_configs/test/build_config_runner_test.dart (hunk 3 above)
Comment (zanderso) [reply tools/pkg/engine_build_configs/test/build_config_runner_test.dart]: Done.

**Pull request merged**